apps/staging_sync/app.py

```diff
@@ -0,0 +1,132 @@
+import os
+import re
+import subprocess
+import sys
+from pathlib import Path
+
+import streamlit as st
+import streamlit.components.v1 as components
+from git.exc import GitCommandError
+from git.repo import Repo
+from rich.ansi import AnsiDecoder
+from rich.console import Console
+from rich.terminal_theme import MONOKAI
+
+from etl.paths import BASE_DIR
+
+CURRENT_DIR = Path(__file__).resolve().parent
+
+
+def main():
+    st.title("etl-staging-sync")
+    st.markdown(
+        """
+    Sync charts and revisions from staging server to production.
+    """
+    )
+
+    # SIDEBAR
+    with st.sidebar:
+        with open(CURRENT_DIR / "instructions.md", "r") as f:
+            st.markdown(f.read())
+
+    st.markdown("### Config")
+    source = st.text_input(
+        "Source",
+        placeholder="my-branch",
+        help="Name of the branch to sync from (with existing `staging-site-mybranch` server).",
```

Comment:
I am not sure I understand this.

Does this mean that the branch in ETL should have its corresponding staging server created, otherwise this won't work?

As I was reading, I was wondering when this is not the case. As long as I create a PR and wait, the corresponding server should exist, shouldn't it?

Maybe a clarification here could be helpful. Or a link to our docs if we have a section where we explain how we create staging servers from PRs.

Reply:
> As long as I create a PR and wait, the corresponding server should exist, shouldn't it?

That's right. I tried to rephrase it to make it more clear. If you find a better wording, feel free to update it, please.